src/mtev_net_heartbeat.c

```diff
@@ -152,15 +204,30 @@ mtev_net_heartbeat_handler(eventer_t e, int mask, void *closure, struct timeval
     evp_ctx = EVP_CIPHER_CTX_new();
     EVP_CipherInit(evp_ctx, EVP_aes_256_cbc(), ctx->key, ivec, false);
     mtevAssert(EVP_CIPHER_CTX_iv_length(evp_ctx) == HDR_IVSIZE);
-    EVP_DecryptUpdate(evp_ctx,text,&outlen1,
-                      (unsigned char *)payload,len);
-    EVP_DecryptFinal(evp_ctx,text+outlen1,&outlen2);
+    if (!EVP_DecryptUpdate(evp_ctx,text,&outlen1,
+                           (unsigned char *)payload,len)) {
+      char addr_str[INET6_ADDRSTRLEN];
+      get_ip_addr_from_sockaddr((struct sockaddr *)&peer_addr, addr_str, sizeof(addr_str));
+      ERR_print_errors_cb(log_ssl_decrypt_error, &addr_str);
+      continue;
+    }
+    if (!EVP_DecryptFinal(evp_ctx,text+outlen1,&outlen2)) {
```

Comment:
Can you combine this into a single if statement as the body is the same?

if(!Update || !Final) {....}

Reply:
Done